infra/gcp/ensure-release-projects.sh

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env bash
+#
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script is used to ensure Release Managers have the appropriate access
+# to SIG Release GCP projects.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SCRIPT_DIR=$(dirname "${BASH_SOURCE[0]}")
+. "${SCRIPT_DIR}/lib.sh"
+
+function usage() {
+    echo "usage: $0 [repo...]" > /dev/stderr
+    echo "example:" > /dev/stderr
+    echo "  $0 # do all staging repos" > /dev/stderr
+    echo "  $0 coredns # just do one" > /dev/stderr
+    echo > /dev/stderr
+}
+
+# NB: Please keep this sorted.
+PROJECTS=(
+    k8s-staging-release-test
+    k8s-release-test-prod
+)
+
+if [ $# = 0 ]; then
+    # default to all staging projects
+    set -- "${PROJECTS[@]}"
+fi
+
+ADMINS="release-managers-admins@kubernetes.io"
+WRITERS="release-managers-private@kubernetes.io"
+VIEWERS="release-managers@kubernetes.io"
+
+for REPO; do
+    color 3 "Configuring: ${REPO}"
+
+    # The GCP project name.
+    PROJECT="${REPO}"
+
+    # The names of the buckets
+    STAGING_BUCKET="gs://${PROJECT}" # used by humans
```

Comment:
This is the piece I want to discuss most - Do we want these projects "governed" by 2 ensure scripts?  This overlaps directly with the other scripts, which means it will drift.

There's a particular set of semantics that we ascribe to a "staging project" and a "prod project".  You are adding on to that, clearly, but it's not clear if you are changing more than that.  In other words - should we make a clean break and remove these release projects from the other scripts?  Or should we build on those?

For example, this script could do something like:


```
# First ensure that the project is a normal staging project
"${ROOT}/ensure-staging-project.sh" release-test
# Now apply customizations for release staging
```

If it helps, we can refactor the existing scripts to make this more obvious.  I knew this was likely to come up, so we it should not be hard - we left room.